File: ChiaMonitor/app/src/main/java/com/example/chiamonitor/data/mappers/BlockchainState.kt
package com.example.chiamonitor.data.mappers

import com.example.chiamonitor.data.remote.blockchain_state.BlockchainStateDto
import com.example.chiamonitor.domain.BlockchainState


fun BlockchainStateDto.toBlockchainState(): BlockchainState {
    return BlockchainState(
        nodeId = node_id,
        height = height,
        blockMaxCost = block_max_cost,
        difficulty = difficulty,
        space = space,
        syncing = syncing,
        syncProgressHeight = sync_progress_height,
        syncTipHeight = sync_tip_height,
    )
}

File: chia-rest/app/entities/mappers/blockchain_state.py
"""Block chain state related mappers."""


from app.entities.blockchain_state import BlockChainState


def parse_rpc_response_to_blockchain_state(rpc_response: dict) -> BlockChainState | None:
    try:
        return BlockChainState(
            node_id=rpc_response['node_id'],
            height=rpc_response['peak']['height'],
            block_max_cost=rpc_response['block_max_cost'],
            difficulty=rpc_response['difficulty'],
            space=rpc_response['space'],
            syncing=rpc_response['sync']['sync_mode'],
            sync_progress_height=rpc_response['sync']['sync_progress_height'],
            sync_tip_height=rpc_response['sync']['sync_tip_height'],
        )
    except (KeyError, TypeError):
        return None
